ui/hooks/src/use-logged-in.ts

```diff
@@ -10,8 +12,50 @@ import { useGetLogin } from './use-login.new';
  */
 export function useLoggedIn() {
   const loginQuery = useGetLogin();
-  const data = useMemo(() => {
-    return loginQuery.data;
-  }, [loginQuery.data]);
-  return { isLoggedIn: !!data?.id, loggedInProfileId: (data?.id || '').toLowerCase() };
+  const queryClient = useQueryClient();
+
+  const myProfileQuery = useGetMyProfileQuery(null, {
+    enabled: !!loginQuery.data?.id,
+    //users who hasnt filled in their account info has no akashaProfile(null), but they are considered logged in if the viewer is not null
+    select: response => response.viewer,
+  });
+
+  console.log('useLoggedIn', 'useGetLogin', loginQuery, myProfileQuery);
+
+  useEffect(() => {
+    if (loginQuery.data?.id && !myProfileQuery.data) {
+      console.log('first condition true');
+
+      const refetchProfileData = async () => {
+        const { data } = await myProfileQuery.refetch();
+        return data;
+      };
+
+      if (!refetchProfileData()) {
+        queryClient.setQueryData([LOGIN_STATE_KEY], null);
+      }
```

Comment:
```suggestion
      const refetchProfileData = async () => {
            await myProfileQuery.refetch();
      };
      
      refetchProfileData();
```

I am not seeing the need to do `queryClient.setQueryData([LOGIN_STATE_KEY], null);`. And if this is needed then refetchProfileData() is still a promise as you have wrote it. Either you need an `IIFE` or move the `setQueryData` logic inside the `refetchProfileData` function.

Reply:
I use this one `queryClient.setQueryData([LOGIN_STATE_KEY], null)` for the case when I click on connect immediately after logging out and the `LOGIN_STATE_KEY` just switched from null to logged in data automatically.  But I'm removing this entire logic and testing if it still works or not on the PR preview.

Reply:
The refetch is needed but not `queryClient.setQueryData([LOGIN_STATE_KEY], null)`.